### PR TITLE
Some refinements to the subscription

### PR DESCRIPTION
**Pull request by lakinwecker**
Adding an event should be far easier now. :)

### CODE REVIEW

**Comment by lakinwecker** [comment]
Also fixes some ratings related bugs

File: src/slack.js [hunk 1]
```diff
@@ -318,6 +373,27 @@ function on(options, callback) {
         }));
     });
 }
+//------------------------------------------------------------------------------
+// A helper method to workaround a bug in botkit. 
+//
+// We store a reference to 'bot' in the startRTM method and then use it here
```
Comment (endrawes0) [inline src/slack.js]: Sneaky

File: src/subscription.js [hunk 2]
```diff
@@ -249,24 +244,43 @@ function processRemoveSubscriptionCommand(config, message, id) {
...
 //------------------------------------------------------------------------------
+// Register an event + message handler
+//------------------------------------------------------------------------------
+function register(bot, event_name, cb) {
+    // Ensure this is a known event.
+    events.push(event_name);
+    
+    // Handle the event when it happens
+    emitter.on(event_name, function(league, sources, context) {
+        return getListeners(league.options.name, sources, 'a-game-is-scheduled').then(function(targets) {
```
Comment (endrawes0) [inline src/subscription.js]: Should this be `event_name` not a hard coded event?

Comment (endrawes0) [inline src/subscription.js]: The hard coded event name it's my only concern here.

**Comment by endrawes0** [comment]
The hard coded event name it's my only concern here.

File: src/subscription.js (hunk 2 above)
Comment (lakinwecker) [inline src/subscription.js]: Yes, it should be!  Good catch.

**Pull request merged**